libs/gnutls/Makefile

```diff
@@ -6,13 +6,13 @@
 include $(TOPDIR)/rules.mk
```

Comment:
You should rebase your branch to avoid to have this change again since it was merged to the master branch already.

Reply:
I understand the problem ("it"(?) was merged to the master == libs/gnutls/Makefile?) but not how to resolve the problem (or that it needs resolving):

```
skyper@spice:~/research/openwrt/packages(master) »git pull
Already up to date.
skyper@spice:~/research/openwrt/packages(master) »git rebase
Current branch master is up to date.
```

```
skyper@spice:~/research/openwrt/packages(master) »git log --graph --oneline | head -n3
* 03240d90f (HEAD -> master, origin/master, origin/HEAD) gsocket: upstream update to 1.4.37
* 6385813dd gnutls: update to 3.7.5
* 891cfc866 gnutls: cleanup Makefile
```

I can delete my fork and start again? That seems to be the easiest(?).

thank you for your help.

Reply:
I don't think that what si proposed as the easiest solution would be sufficient, because you will create each time a new PR once you did not do correctly? :-) I will write you steps, which I did during afternoon/evening. Right now, I will fix it for you.

Reply:
Hmm. Commit 5d72b6abf348f42a9b1ce11a456307f709b78dd1 failed the formality check. I do not know why this commit made it into the pull request at all (it was done after the pull request and after approval by neheb). Hours after the 'pull request' the webpage github.com notified me that the 'master is 2 commits ahead' and suggested that I should 'merge' the changes from the master into my fork to stay current.

Made sense to me and so I did.

I guess doing this on github.com triggered another commit for an earlier pull? Can you help?

I'm happy to restart again from new. It's just a 'make clean openwrt' on my system and takes no time or effort at all. It deletes everything and starts from a clean checkout and slate.

Reply:
What I did for you yesterday and today. I'm using CLI and SSH.

1. First, you need to clone your repository to your computer to some folder. In my case, I created a folder ``skyper`` and cloned your repository files to it.
```
git clone git@github.com:SkyperTHC/packages.git skyper
```

2. Check out the branch, which you want. Usually, it would help if you created a feature branch and from that branch send a pull request. Since you did your changes in the master branch, this step can be skipped.

```
git checkout master
```

3. Add [remote](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes)

```
git remote add upstream git@github.com:openwrt/packages.git
```

4. Since we added a remote repository, we need to fetch it first.

```
git fetch upstream
```

5. Now, we will rebase your branch on top of the remote branch to remove the merge commit. The parameter [rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) is important. If you do not have it, it will create a merge commit like it happened if you did it via WebUI.

```
git pull upstream master --rebase
```

6. We need to force push your local changes to your's remote, in this case, to your repository on GitHub. We can not simply push it as your remote is different. Be careful with force pushing. It would be best if you used it with [--lease](https://www.freecodecamp.org/news/git-please-a182f28efeb5/)

```
git push --force
```

There is also simplified parameter ``-f``, which I am using, but for this know-how, it is better to use it in full because you wouldn't know what it means.

If you do something wrong and then create a new pull request, it is not a solution. What would you do if someone asked you to make more changes to your commit? Will you close it and create a new one? No. :) Because Git as a version control system can be very useful if you know what you should do. There is also [git commit --amend](https://www.atlassian.com/git/tutorials/rewriting-history), and I think this one can be useful for you as well.

To your question, that's not related to the OpenWrt build system, and also it is now up to you if you do any changes.

Reply:
Thank you for this very helpful and  enlightening explanation. Is there anything else I have to do at this stage or just “sit and wait” until the pull request is merged?